### PR TITLE
Necropolis, The Sunken City Pt 2

### PR DESCRIPTION
**Pull request by acidia**
This introduces a few NPCs into the vault proper.  

Now necropolis is mentioned after the final mission of the refugee shelter's old guard mission is complete.  The tip isn't necessary to fully explore the content that is added though.  What you do need is to be designated a federal marshal in order for the two quest givers to open up to you.

The Captain in charge of the vault tasks you to locate the commo team that went to secure the communications array.  Find the team and return with a list of frequencies to complete the mission.  Next the captain will request that you cull a number of the charred nightmares that inhabit the vault.

The commo guy has his own quest for you to build him a unique electronics part, the radio repeater mod.  It isn't too hard but gathering the parts will take a little bit.  Next up he needs you to separate the vault grid from the local power grid.  This is a slight hangup since jsons don't have computer support yet.  You can complete the mission currently by activating the correct command on a radio stations terminal.  You will then be informed that shorty he will need you to go on missions to connect various surface radio stations into the emergency repeater network by constructing the part and installing it into their main terminal.  This is now an ongoing mission...

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
This all sounds great!

**Comment by KA101** [comment]
Hmm.  I'd thought that folks like the OG rep might eventually hand out map-random missions, such as "escort these refugees to that farm that we believe is abandoned over there, so they can get it working again", for example.

But this certainly works too.  :-)

**Comment by acidia** [comment]
Ya, the captain elaborates a bit on their mission... long term goal is to clear the vault and reactivate.  Eventually it wouldn't be hard to drop off people to create a kind of town.  So a continuing mission would be, "Rescue people by bringing them to me" and the reward would "Hey the person is converted into a static npc and given a job" (in reality it would just be deleting the original person and creating a new one with the same name but meh)  It makes it feel like you built a town.

File: data/json/items/books.json [hunk 1]
```diff
@@ -3125,6 +3125,50 @@
     },
     {
         "type" : "BOOK",
+        "id" : "necropolis_freq",
+        "name" : "frequency list",
+        "name_plural" : "Frequency Lists",
```
Comment (Lain-) [inline data/json/items/books.json]: Not sure why the singular form is "frequency list" but the plural form is "Frequency Lists". I assume it's an oversight?

**Comment by Snaaty** [comment]
Awesome! Also really good to see some quest-code.

**Comment by KA101** [comment]
Hmm.  I've not looked at this thanks to the BUG flag.  Something up?

**Comment by BevapDin** [comment]
acidia is most likely referring to #11667

**Comment by acidia** [comment]
Ok, added the follow on quest for the commo guy.  Computers can now complete predetermined missions and radio stations are standing by to be 'upgraded' with the repeater mod.  Despite what quest number 2 says, I currently have the power disconnect on radio stations pending a pull request to add computer support for json buildings.  (I can't tell who e-mail me since it used their real name instead of their user name)

Yup, I was kinda referring to #11667 but when I did the fix myself it left the game prone to crashing near static spawns... could there be something else?

**Comment by BevapDin** [comment]
The other bug could have been fixed by #11638, I did encounter the bug in your branch. However, with the commit from the mentioned PR (already in the master branch), everything works fine fine.

**Comment by acidia** [comment]
Ah, then we should be good to go!

File: src/mission_start.cpp [hunk 2]
```diff
@@ -252,6 +252,16 @@ void mission_start::kill_100_z(mission *miss)
...
 }
 
+void mission_start::kill_20_nightmares(mission *miss)
+{
+ npc *p = g->find_npc(miss->npc_id);
+ p->attitude = NPCATT_FOLLOW;//npc joins you
+ miss->monster_type = "mon_charred_nightmare";
+ int killed = 0;
+ killed += g->kill_count("mon_charred_nightmare");
+ miss->monster_kill_goal = 20+killed;//your kill score must increase by 100
```
Comment (KA101) [inline src/mission_start.cpp]: Surely 20?

File: src/npctalk.cpp [hunk 3]
```diff
@@ -849,6 +849,80 @@ std::string dynamic_line(talk_topic topic, npc *p)
...
+            case 4:
+                if (g->u.male)
+                    return _("Sir.");
+                else
+                    return _("Ma'am");
+            case 5:
+                if (g->u.male)
+                    return _("Dude, if you can hold your own you should look into enlisting.");
+                else
+                    return _("Hey miss, don't you think it would be safer if you stuck with me?");
```
Comment (KA101) [inline src/npctalk.cpp]: We can discuss sexism in survival situations later.  /Ellie Sattler FTW

**Comment by KA101** [comment]
Couple things that can be fixed in another PR:
1) The SFC calls xyr CO male, but CPT Joan Wagoner sounds like a feminine name.  Might want to watch out there.
2) Debug messages come up for the kill count.  Not urgent to change that, but worth noting.
2a) The CPT doesn't say she's going to join me for the patrol, so that might be a surprise for folks.  Watch where you point that M2, KA101!
3) The SFC should probably give you back the radio repeater mod the first time xe sends you to install one, unless xe needs it for the vault (and if so, I missed that statement).

This is good to go, holding on the build failures ATM.

**Comment by acidia** [comment]
The next level is done excepts for the item fields.  If BevapDin's Mapgen 2 commands work the way I hope, I should be able to push the mentioned fixes and the next level tomorrow.

**Comment by Rivet-the-Zombie** [comment]
Nice!

**Pull request merged**